src/inmanta/ast/statements/generator.py

```diff
@@ -315,8 +342,33 @@ def execute(self, requires: Dict[object, object], resolver: Resolver, queue: Que
         # the type to construct
         type_class = self.type.get_entity()
 
+        # kwargs
+        direct_kwarg_attrs: Dict[str, InmantaType] = {}
+        indirect_kwarg_attrs: Dict[str, ExpressionStatement] = {}
+        for kwargs in self.wrapped_kwargs:
+            for (k, v) in kwargs.execute(requires, resolver, queue):
+                if k in self.attributes or k in direct_kwarg_attrs or k in indirect_kwarg_attrs:
+                    raise RuntimeException(
+                        self, "The attribute %s in the constructor call of %s is already set." % (k, self.class_type)
```

Comment:
when reporting  is already set, I would like it to be more precise of where it was already set
e.g. 
``The attribute %s is set twice in the constructor call of %s ```

even beter would be to report where both values come from, but that is perhaps too much to take in this ticket.
(cfr the DoubleSetException)

Reply:
I processed this. Should I make a ticket for more informative reporting before marking this as resolved?

Reply:
that would be good, but mark it as 'enhancement'

Reply:
#1713